fast64_internal/oot/scene/exporter/to_c/scene_table_c.py

```diff
@@ -158,6 +171,25 @@ def getDrawConfig(sceneName: str):
     raise PluginError(f"Scene name {sceneName} not found in scene table.")
 
 
+def addHackerOoTData(fileData: str):
+    """Reads the file and adds HackerOoT's modifications to the scene table file"""
+    newFileData = '#include "config.h"\n\n'
+
+    for line in fileData.split("\n"):
+        if "// Debug-only scenes" in line:
+            newFileData += "\n#ifdef INCLUDE_TEST_SCENES\n"
```

Comment:
this is very inefficient (performance wise)
when you do `strA + strB`, it copies strA and strB to new memory for the resulting string
it's fine for a few uses, but doing `+=` in big loops like here means the concatenation ends up being of quadratic complexity
instead:
use a `lines = []` list, append each line to it individually, and return `"".join(lines)` or `"\n".join(lines)` which achieves linear complexity

Reply:
so, instead of ``file.read()`` use readlines and then go through the list to do the same thing (except you replace the thing from the list of course)

and return with a str.join?

Reply:
well you could just change `newFileData = []` and `newFileData += s` -> `newFileData.append(s)` `return "".join(newFileData)`

I suggested `lines` because it's easier to grasp but it doesn't need to be one line per string in that list